src/js/BookReader.js

```diff
@@ -1584,44 +1584,61 @@ BookReader.prototype.isFullscreen = function() {
   return this.isFullscreenActive;
 };
 
-BookReader.prototype.toggleFullscreen = function() {
-  if (this.isFullscreen()) {
+/**
+ * Fullscreen mode manager
+ * - toggles fullscreen
+ * - binds keyboard controls
+ * - fires custom event
+ * @param { boolean } bindKeyboardControls
+ */
+BookReader.prototype.toggleFullscreen = function(bindKeyboardControls = true) {
+  this._fullscreenCloseHandler = function _fullscreenCloseHandler(e) {
+    if (e.keyCode === 27) this.toggleFullscreen();
+  }.bind(this);
+  const handleExit = () => {
     this.exitFullScreen();
-  } else {
+    $(document).unbind('keyup', this._fullscreenCloseHandler);
+  }
+  const handleEnter = () => {
     this.enterFullscreen();
+    if (bindKeyboardControls) {
+      $(document).keyup(this._fullscreenCloseHandler);
+    }
+  }
+
+  if (this.isFullscreen()) {
+    handleExit();
```

Comment:
These two new functions, `handleExit` and `handleEnter` are a little confusing since there are already `enterFullscreen` and `exitFullScreen` functions. Can we dissolve them and place the keyup bind/unbind in `enterFullscreen` and `exitFullScreen`? Also, that code should be run whenever full screen is entered; not just when it's toggled.